packages/hub/src/lib/cache-management.ts

```diff
@@ -0,0 +1,257 @@
+import { homedir } from "node:os";
+import { join, basename } from "node:path";
+import { stat, readdir, readFile, realpath, lstat } from "node:fs/promises";
+import type { Stats } from "node:fs";
+
+function getDefaultHome(): string {
+	return join(homedir(), ".cache");
+}
+
+function getDefaultCachePath(): string {
+	return process.env["HF_HOME"] ?? join(process.env["XDG_CACHE_HOME"] ?? getDefaultHome(), "huggingface")
+}
+
+function getHuggingFaceHubCache(): string {
+	return  process.env["HUGGINGFACE_HUB_CACHE"] ?? getDefaultCachePath();
+}
+
+function getHFHubCache(): string {
+	return  process.env["HF_HUB_CACHE"] ?? getHuggingFaceHubCache();
+}
+
+const FILES_TO_IGNORE: string[] = [".DS_Store"];
+
+export enum REPO_TYPE_T {
+	MODEL = "model",
+	DATASET = "dataset",
+	SPACE = "space",
+}
+
+export interface CachedFileInfo {
+	filename: string;
+	filePath: string;
+	blobPath: string;
+	sizeOnDisk: number;
+
+	blobLastAccessed: number;
+	blobLastModified: number;
+}
+
+export interface CachedRevisionInfo {
+	commitHash: string;
+	snapshotPath: string;
+	sizeOnDisk: number;
+	readonly files: Set<CachedFileInfo>;
+	readonly refs: Set<string>;
+
+	lastModified: number;
+}
+
+export interface CachedRepoInfo {
+	repoId: string;
+	repoType: REPO_TYPE_T;
```

Comment:
There are already `RepoId` and `RepoType` types in `packages/hub/src/types/public.ts`

```suggestion
	id: RepoId;
```

Reply:
I think you missed this one :)

Reply:
Sorry !! Just committed the change